main.go

```diff
@@ -88,6 +98,31 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Get STS vars
+	// ROLEARN env var is set via operator subscription
+	var roleARN string
+	if common.CCOWorkflow() {
+		roleARN = os.Getenv("ROLEARN")
```

Comment:
Do we want to add a check for empty role arn and expose that to the user for fixing? 

Is there a doc of where the CSV can be updated if you do mess this up when installing?

Reply:
Role ARN is a compulsory field when the ROSA cluster is in  STS mode, you won't be able to install the operator without supplying a role arn during install time.

Reply:
Are there other situations where someone may have this cluster in STS mode, want to use this feature and not have this set? 

I wonder if it is worth the extra effort, just in case?

Reply:
The `common.CCOWorkflow` call checks if ENV VAR exists @shawn-hurley , so this only triggers if env var is set

was this your question?

Reply:
It makes sense that we have empty roleARN sometimes. 

But there is also a distinct possibility that the cred-request will get triggered with an empty ARN:

https://github.com/openshift/oadp-operator/pull/1106/files/8f1850453cbdc824542133a6d0fdeddb0c10ed59#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R110-R124

I assume that this will lead to some issues. 

We need to make it clear a couple of things:

1. When a user should have put this value in here, but it is not, then we need to tell them. 
2. When a user should not have put this value in here, then we probably don't want to create the CredentialsRequerst correct?

Reply:
@shawn-hurley cred-request only gets created if the role arn is supplied https://github.com/openshift/oadp-operator/pull/1106/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R107-R127

Reply:
something just changed, this was not the code I was looking at. why do you have this then?
https://github.com/openshift/oadp-operator/pull/1106/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R106

Reply:
yeah I can remove that var and update the PR, is there anything else that I need to update ?

Reply:
@shawn-hurley Done, added logs, updated the PR, PTAL thanks!